### ISSUE TITLE
Upgrade to benchmarking interval

### ISSUE THREAD

**Issue by thebeerbaron2018** [MultiPoolMiner/MultiPoolMiner#2026]
Would it be possible to have two intervals.  One "Interval" as the time between cycles and "BenchmarkInterval" which is used to benchmark algorithms.  The reason for my request is that ZergPool decided to add/change Groestl today.  As a result all of my machines decided to (re)benchmark.  There are 8 miners supporting Groestl and I have the switching interval set to 1 hour.  This means my entire farm has benchmarked Groestl (which isn't very profitable) for 8 hours!

**Comment by m-bosch** [comment]
why would you set your $interval setting to 3600? its by default 60 (1minute). If im correct MPM monitor won't see if you are online, only once a hour?

**Comment by thebeerbaron2018** [comment]
@UselessGuru  I set interval to 3600 because I find switching only every hour is more profitable for me on PPLNS due to less switching the shares are worth more if you continue to put into the pot as it were.... the problem occurred today because I don't use an algorithm list - Zergpool added Groestl which had never been benchmarked by any of my rigs.  Normally I benchmark on one reserved machine with the 600 interval and then copy the stats files over to the farm of 200 odd machines.  When Zerg added the algo it was seen as new by ALL the rigs as none of them had benchmark stats for it.... so they all started benchmarking it.... now there are 8 miners in my collection that can mine Groestl and the interval is set to.... yup... you guessed it ... 3600 - so 8 x 3600mins of benchmarking ON EVERY RIG!!!!!! hahaha it actually made me laugh!  What I was kinda requesting would be so I could set "-interval 3600 -benchinterval 600" so that it will use the shorter cycle for algos that pop up out of the blue.  Finally yes I know this could have easily been solved by using an algorithm list but I like to be dynamic so if new coins are added and I have miners available that they jump on it without intervention, benchmark and then min if profitable - hope that makes sense?

**Comment by grantemsley** [comment]
At least it wasn't X16R...with your interval set that high, it would have benchmarked for 10 HOURS.

But we don't recommend setting the interval that high.  The supported way is to increase SwitchingPrevention, which is a multiplier for the price bias of the pool.  That bias carries over to the miners, making the miners that aren't running appear to be a little less profitable, and the miners that are running stay the same.  That's supposed to prevent it from switching back and forth between two algorithms that are almost the same profit.

You can see the effect that has in the web interface - go to the miners tab, expand a miner with the plus sign on the left, and look at Profits_Bias and Profits_Unbias.    Profits_Unbias is the actual projected profit.  Profits_Bias is the reduced profitability used in the switching calculations.  On running miners, Profits_Bias is set to be Profits_Unbias, so they don't get reduced, and are preferred if two algorithms are nearly the same profits.

**Comment by UselessGuru** [comment]
@m-bosch 

I am talking $interval

$Delay defaults to 0. It may be neccesary to increase thie to a few seconds on unstable systems (see readme)

**Comment by thebeerbaron2018** [comment]
I was thinking about the extended interval actually as you wrote that...  so if there were 8 miners that did X16R and my interval was an hour it would have actually ended up with 80 hours of benching!!! lol. I've never been able to get switching prevention to appear to work as described (something I note a lot of other people saying too) and changing the value from 1 to 2 to 10 to 50 doesn't appear to make that much difference for me which is why I use the hour interval and just have at most 24 miner swaps a day or less if something is profitable across hours.  I've tested 5 rigs in parallel with prevention set to 1,2 and 10 (interval on 120 seconds) and then the fourth rig I used no switching prevention and set the interval to 3600 and the fifth rig was set to prevention of 5 and interval 3600..... they all ran 4x1070's for a period of 4 days (96 hours) and the result for me was rig four the winner.
